src/Shared/SizeLimitedStream.cs

```diff
@@ -48,7 +49,14 @@ public override int Read(byte[] buffer, int offset, int count)
         _totalBytesRead += bytesRead;
         if (_totalBytesRead > _sizeLimit)
         {
-            throw new InvalidOperationException("The maximum number of bytes have been read.");
+            if (_handleSizeLimit != null)
+            {
+                _handleSizeLimit(_totalBytesRead, _sizeLimit.Value);
+            }
+            else
+            {
+                throw new InvalidOperationException("The maximum number of bytes have been read.");
```

Comment:
Should this message include `_sizeLimit.Value`, so it's aids diagnosing the error?

Reply:
"The maximum number of bytes have been read." is the original message, i did not change it